### PR TITLE
Story/VSPC-101

### PR DESCRIPTION
**Pull request by prjadhav14**
If the choice lists in the slide contain sequences with other choices creation of choice block fails with stack overflow log as choices list has sequences and sequences has choices in them.

Fixed by annotating mapping of sequence with JsonIgnore

### CODE REVIEW

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
there is more wrong with this. If I dont' select the "all" checkbox, but select individual choices, I don't see a new choice block after I submit. reloading the page does show the choice block but out of order (not at the end of the block list).

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddChoiceBlockController.java [hunk 1]
```diff
@@ -22,12 +25,24 @@
...
+        List<Object> responseData = new ArrayList<>();
+        responseData.add(choiceBlock);
+        if(!choiceBlock.isShowsAll()) {
+            List<ISequence> selectedSequences = new ArrayList<ISequence>();
+            for(IChoice choice:choiceBlock.getChoices()) {
+                selectedSequences.add(choice.getSequence());
+            }
+            responseData.add(selectedSequences);
+        }
+        return new ResponseEntity<>(responseData, HttpStatus.OK);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddChoiceBlockController.java:46]: I would return a map here and properly label the returned values

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddChoiceBlockController.java [hunk 2]
```diff
@@ -22,12 +25,24 @@
...
+        /*After annotating sequence attribute with JsonIgnore in Choice model to fix stack overflow issue 
+         * the sequences are not returned as part of choiceBlock object, hence sequences are explicitly passed 
+         * as part of Response in a list form*/
+        List<Object> responseData = new ArrayList<>();
+        responseData.add(choiceBlock);
+        if(!choiceBlock.isShowsAll()) {
+            List<ISequence> selectedSequences = new ArrayList<ISequence>();
+            for(IChoice choice:choiceBlock.getChoices()) {
+                selectedSequences.add(choice.getSequence());
+            }
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddChoiceBlockController.java:43]: use map, collect here.

**Comment by prjadhav14** [comment]
> Congratulations, build succeeded!

Mistakenly, Added the wrong id in the commit message.

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Pull request merged**